app/views/assignments/_assignment.html.slim

```diff
@@ -7,12 +7,13 @@ tr
   - if action_name == 'index'
     td = link_to assignment.volunteer, assignment.volunteer
   td= t("simple_form.options.assignment.state.#{assignment.state}") if assignment.state.present?
-  td= l(assignment.created_at, format: :date_only)
+  td= l(assignment.period_start) if assignment.period_start
+  td= l(assignment.period_end) if assignment.period_end
   td
     - if policy(User).show?
-      = link_to assignment.creator, assignment.creator if assignment.creator.present?
+      = link_to assignment.creator.full_name, profile_path(assignment.creator.profile) if assignment.creator.present?
     - else
-      = assignment.creator
+      = assignment.creator.full_name
```

Comment:
`assignment.creator.full_name` on the other hand could fail on edge case, because `User` could have its Profile record missing.
I'd suggest `assignment.creator&.full_name` or even `assignment.creator&.profile&.contact&.full_name` because I wouldn't be 100% sure how this delegate on `User` acts in that case.

Reply:
Based on how `full_name` is defined, wouldn't it load the email if the user has no profile?

`def full_name`
`  if profile&.contact`
`    "#{profile.contact.last_name}, #{profile.contact.first_name}"`
`  else`
`    email`
`  end`
`end`